frontend/packages/kubevirt-plugin/src/components/dashboards-page/overview-dashboard/activity.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { ActivityProgress } from '@console/shared/src/components/dashboard/activity-card/ActivityItem';
+import { ResourceLink } from '@console/internal/components/utils';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { VirtualMachineModel } from '../../../models';
+
+export const DiskImportActivity = ({ resource }) => (
+  <ActivityProgress
+    title="Importing VM disk"
+    progress={parseInt(_.get(resource, 'status.progress', 0), 10)}
+  >
+    <ResourceLink
+      kind={referenceForModel(VirtualMachineModel)}
+      name={resource.metadata.ownerReferences[0].name}
+      namespace={resource.metadata.namespace}
+    />
+  </ActivityProgress>
+);
+
+export const V2VImportActivity = ({ resource }) => (
+  <ActivityProgress
+    title="Importing VM (v2v)"
+    progress={parseInt(_.get(resource.metadata.annotations, 'v2vConversionProgress', 0), 10)}
```

Comment:
can `resource` be undefined?

Reply:
It cannot. This is a resource which is fetched from k8s (definition is https://github.com/openshift/console/pull/3577/files#diff-ebd2d0d608781a3c30e4c97813f813c2R284 ), run through `isActivity` for given activity like https://github.com/openshift/console/pull/3577/files#diff-ebd2d0d608781a3c30e4c97813f813c2R289 or https://github.com/openshift/console/pull/3577/files#diff-ebd2d0d608781a3c30e4c97813f813c2R306 which if returns true, then the resource is passed to component.